src/constraints/constraint_units_available.jl

```diff
@@ -53,7 +53,9 @@ function _build_constraint_units_available(m, u, s, t)
             init=0,
         )
         <=
-        + number_of_units(m; unit=u, stochastic_scenario=s, t=t)
+        # Change the default number of units so that it is zero when candidate units are present
+        # and otherwise 1.
+        + ifelse(is_candidate(unit=u), 0, number_of_units(m; unit=u, stochastic_scenario=s, t=t) )
```

Comment:
Doesn't this disallow having both number_of_units and candidate_units different than zero? It looks like if the user specifies candidate_units different that zero then any non zero value for number_of_units is just ignored? Or I am missing something?

Reply:
Indeed. Now I tried to correct it.